### ISSUE TITLE
[Feature Request] Improve usability in Shortcut mapper: Make the columns resizable

### ISSUE THREAD

**Issue by rillig** [notepad-plus-plus/notepad-plus-plus#5610]
Extracted from #2563:

## Make the columns resizable

When the columns are resizable, there is no need for the horizontal scrollbar to be shown by default. Initially the Shortcut column should be as wide as necessary, and the Name column should take the rest of the space.

**Comment by sasumner** [comment]
It's an okay suggestion...my workaround is to just drag the window bigger in both dimensions by grabbing a corner.

**Comment by eljefe7000** [comment]
I can make the columns resizable, but I'm unsure about the initial column size.  It sounds like you want the Name and Shortcut columns to be visible, but not the Category or the Plugin column.  Is that what you mean?

**Comment by rillig** [comment]
No, it's not what I meant back then. When I wrote the issue, I did not see the Category or the Plugin column, otherwise I would have mentioned them.

* Initially the Shortcut column should be completely visible since it is important to see the complete shortcut.
* The Category column should be the next one that gets as much space as necessary since I expect that category names are usually short, at least shorter than the Name column.
* The Name column should get the remaining space since it can get quite long.

So basically there is a precedence between the columns, and whichever column comes first gets the space it needs. The Shortcut column has highest precedence, the Name column has lowest precedence, the other columns are somewhere in between.

**Comment by eljefe7000** [comment]
@rillig 

Can you split this into two requests, one for making the columns resizable and one for setting the default sizes?  I have already created the code to make the columns resizable and the default sizing will require more research.
